src/OrchardCore.Modules/OrchardCore.Markdown/Startup.cs

```diff
@@ -22,14 +24,27 @@ namespace OrchardCore.Markdown
 {
     public class Startup : StartupBase
     {
+        private readonly IShellConfiguration _shellConfiguration;
+
         static Startup()
         {
             TemplateContext.GlobalMemberAccessStrategy.Register<MarkdownBodyPartViewModel>();
             TemplateContext.GlobalMemberAccessStrategy.Register<MarkdownFieldViewModel>();
         }
 
+        public Startup(IShellConfiguration shellConfiguration)
+        {
+            _shellConfiguration = shellConfiguration;
+        }
+
         public override void ConfigureServices(IServiceCollection services)
         {
+            var markdownOptionsConfig = _shellConfiguration.GetSection("OrchardCore_Markdown");
+            var markdownOptions = new MarkdownOptions();
+            markdownOptionsConfig.Bind(markdownOptions);
+
+            services.Configure<MarkdownOptions>(markdownOptionsConfig);
```

Comment:
We don't need this line if we only use the options to configure the markdown pipeline below

If so we could just do in one line

    var markdownOptions = _shellConfiguration.GetSection("OrchardCore_Markdown")
        .Get<MarkdownOptions>() ?? new MarkdownOptions();

Reply:
We actually shouldn't have a `MarkdownOptions` class at all here, because it is being passed directly into the `.ConfigureMarkdownPipeline` and captured there. This means that `services.Configure<MarkdownOptions>` is unused, because the `IOptions<MarkdownOptions>` is itself never resolved.

In this case it would be better not to have the class at all, and just resolve the string directly from `IShellConfiguration` 

e.g. `_shellConfiguration.GetValue("OrchardCore_Markdown", "nohtml+advanced");`

Reply:
yes indeed if the options is now only a string ;)

Reply:
@deanmarcussen I was think about that before, but this will allow anyone to resolve `IOptions <MarkdownOptions>` instead on working with magic string

Reply:
But it will be useless if resolved because the value of it has already been passed directly into the `MarkdownPipelineOptions`, and that is the only place it is useful.

Reply:
You are right because it's already configured, I will update the PR